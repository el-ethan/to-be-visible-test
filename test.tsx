import React from 'react'
import { render } from '@testing-library/react'
import styled from 'styled-components'
import '@testing-library/jest-dom/extend-expect'

const BasicHiddenDiv = styled.div`visibility: hidden`
const OuterDivWithoutStyles = styled.div``
const OuterDivWithStyles = styled.div`
    #bar {
        color: red;
    }
`

describe('toBeVisible', () => {
    it('works with basic example', () => {
        const {getByTestId} = render(<BasicHiddenDiv data-testid='foo'/>)
        expect(getByTestId('foo')).not.toBeVisible()
    });

    it('works with outer div without styles', () => {
        const {getByTestId} = render(
            <OuterDivWithoutStyles>
                <BasicHiddenDiv data-testid='foo' id='bar'/>
            </OuterDivWithoutStyles>
        )
        expect(getByTestId('foo')).not.toBeVisible()
    });

    it('works with outer div with styles', () => {
        const {getByTestId} = render(
            <OuterDivWithStyles>
                <BasicHiddenDiv data-testid='foo' id='bar'/>
            </OuterDivWithStyles>
        )
        expect(getByTestId('foo')).not.toBeVisible()
    });


    it('renders snapshot with red and hidden', () => {
        const {container} = render(
            <OuterDivWithStyles>
                <BasicHiddenDiv data-testid='foo' id='bar'/>
            </OuterDivWithStyles>
        )
        expect(container).toMatchSnapshot()
    });
});