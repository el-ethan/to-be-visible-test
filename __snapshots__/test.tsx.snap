// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeVisible renders snapshot with red and hidden 1`] = `
<div>
  <div
    class="sc-dlnjwi kesRvd"
  >
    <div
      class="sc-bdnxRM hdqktR"
      data-testid="foo"
      id="bar"
    />
  </div>
</div>
`;
